testmenu.c

```diff
@@ -69,47 +84,83 @@ int main(int argc, char **argv) {
 
-    if( rc != SQLITE_OK ) {
-        fprintf(stderr, "SQL error: %s\n", zErrMsg);
-        sqlite3_free(zErrMsg);
-    }
+    // if( rc != SQLITE_OK )
+    // {
+    //     fprintf(stderr, "SQL error: %s\n", zErrMsg);
+    //     sqlite3_free(zErrMsg);
+    // }
 
-    do {
+    do 
+    {
         // Case 1: User is not connected
-        if(connected == 0) {
+        if(connected == 0)
+        {
             printf("Welcome to your glycemia database, please choose an option:\n");
             printf("1. Log in with your user\n");
             printf("2. Create a new user\n");
-            printf("3. Exit\n");
+            printf("9. Exit\n");
             scanf("%d", &choice);
 
-            if(choice == 1) {
-                printf("Please enter your username:\n");
+            if(choice == 1)
+            {
+                printf("\nPlease enter your username:\n");
                 scanf("%s", &username);
+                strcat(username, "\0");
+                printf("\n");
+
                 printf("Please enter your password:\n");
-                scanf("%s", &password);
-                isUserValid = LogIn(username, password, db); 
-                if(isUserValid) {
-                    printf("You are now connected!\n");
+                fflush(stdin);
+                char* input = malloc(30);
+                char* getPassword = malloc(30);
+                getPassword = fgets(input, 30, stdin);
+                for(int i = 0; i < strlen(getPassword); i++)
+                {
+                    if(getPassword[i] == '\n')
+                    {
+                        getPassword[i] = getPassword [i + 1];
+                    }
+                }
+                printf("\n");
+
+                char sql[200] = "SELECT * FROM USERS WHERE username = '";
+                strcat(sql, username);
+                strcat(sql, "' AND password = '");
+                strcat(sql, getPassword);
+                strcat(sql, "'");
+
+                rc = sqlite3_exec(db, sql, checkUser, 0, &zErrMsg);
+
+                if( rc == SQLITE_OK )
+                {
+                    printf("Wrong username or password, please try again\n\n");
+                    sqlite3_free(zErrMsg);
+                } else
+                {
+                    printf("You are now connected !\n");
+                    printf("Welcome %s ^^\n\n", username);
                     connected = 1;
-                }else {
-                    printf("Wrong username or password, please try again!\n");
                 }
-            }else if(choice == 2) {
-                // sql = "SELECT COUNT * from USERS"
+
+                free(input);
+                free(getPassword);
+
+
+            }else if(choice == 2)
+            {
 
                 char sqltest[200] = "INSERT INTO USERS (USERNAME, PASSWORD, AGE, CREATED_AT) VALUES ('\0";
-                printf("Please enter your username:\n");
+                printf("\nPlease enter your username:\n");
```

Comment:
i would advise u to not mix the scanfs and the strcation of the sql statement. Like, ask for all of the inputs first then do the strcats. It will be more easily readable

Reply:
ok moved it for more clean code